packages/flutter_tools/test/upgrade_test.dart

```diff
@@ -24,5 +37,40 @@ void main() {
       expect(_match('Updating 79cfe1e..5046107'), false);
       expect(_match('229 files changed, 6179 insertions(+), 3065 deletions(-)'), false);
     });
+
+    group('findProjectRoot', () {
+      Directory temp;
+
+      setUp(() async {
+        Cache.disableLocking();
+        context[Logger] ??= new BufferLogger();
+        context[Doctor] = new Doctor();
+        context[DeviceManager] = new MockDeviceManager();
+
+        temp = Directory.systemTemp.createTempSync('flutter_tools');
+        Cache.flutterRoot = '../..';
+        CreateCommand command = new CreateCommand();
+        CommandRunner runner = createTestCommandRunner(command);
+        int code = await runner.run(<String>['create', '--no-pub', temp.path]);
+        expect(code, 0);
+      });
+
+      tearDown(() {
+        temp.deleteSync(recursive: true);
+      });
+
+      test('in project', () {
```

Comment:
Here and in the other `test()`, you might be able to avoid a lot of the context boilerplate in the setup method if you instead use `testUsingContext` (there's an example in `create_test.dart`).